### PR TITLE
Simplify keep_intervals code

### PR DESCRIPTION
**Pull request by hyanwong**
Now we can simply use `delete_sites()` for most of the functionality in `keep_intervals()`, assuming that we want to reduce code duplication at the expense of a (slightly) less efficient algorithm.

### CODE REVIEW

File: python/tskit/tables.py [hunk 1]
```diff
@@ -1780,50 +1780,18 @@ def keep_intervals(self, intervals, simplify=True, record_provenance=True):
...
-        # Don't bother dealing with connected mutations as we are removing the whole site
-        tables.mutations.set_columns(
-            site=site_map[mutations.site[keep_mutations]],
-            node=mutations.node[keep_mutations],
-            derived_state=new_ds,
-            derived_state_offset=new_ds_offset,
-            parent=parent_map[mutations.parent[keep_mutations]],
-            metadata=new_md,
-            metadata_offset=new_md_offset)
+        tables.delete_sites(np.where(~keep_sites)[0], record_provenance=False)
```
Comment (jeromekelleher) [inline python/tskit/tables.py]: I had to look up what the ~ operator does (bitwise complement). I think ``not keep_sites`` is probably easier to read and logically apt.

Comment (hyanwong) [reply python/tskit/tables.py]: Oh, sure. I had forgotten that you could use `not` on a numpy array too.

Comment (hyanwong) [reply python/tskit/tables.py]: Change made, commit amended

**Comment by hyanwong** [comment]
Ah - I didn't test it! you *can't* do `(not np_array)`. I guess I should move to `np.not(np_array)`.

**Comment by jeromekelleher** [comment]
This is behind the tip of master, so it'd be nice to rebase. It doesn't matter in this case (the commit in front is a trivial doc update), but it would be good to rebase off upstream/master as a matter of course when updating PRs.

**Comment by jeromekelleher** [comment]
Merging, thanks.

**Comment by hyanwong** [comment]
> This is behind the tip of master, so it'd be nice to rebase. It doesn't matter in this case (the commit in front is a trivial doc update), but it would be good to rebase off upstream/master as a matter of course when updating PRs.

I try to do that. Sorry I forgot this time. Too many other simultaneous commit pathways in my head!

**Pull request merged**